easybuild/easyblocks/generic/pythonpackage.py

```diff
@@ -704,6 +704,9 @@ def install_step(self):
         # (for iterated installations over multiply Python versions)
         self.install_cmd_output += out
 
+        # fix shebangs if specified
+        self.fix_shebang()
```

Comment:
Won't this pick up a shebang list from the non-extension part of the easyconfig?
That may be a bad thing.

Reply:
Possibly. Do you have a suggestion on how to avoid that ?

Reply:
In reality, I always found it weird that extensions pick up *anything* from the non-extension part...  I've been bitten by it more than once using `configopts` and other parameters.

Reply:
@boegel you should know the answer, does fix_shebang here pick up things from the non-extension part?

Reply:
Yes, if `fix_python_shebang_for` (or another `fix_*_shebang_for`) is set for the "parent", then it will be picked up for all extensions as well.

All easyconfig parameters set for the parent are inherited by each extension, which is important in some cases (like `toolchain` for example). There are some exceptions though, see https://github.com/easybuilders/easybuild-framework/blob/95bae20337781a06defe467faf487a122aae8512/easybuild/framework/extension.py#L108

and https://github.com/easybuilders/easybuild-framework/blob/95bae20337781a06defe467faf487a122aae8512/easybuild/framework/extensioneasyblock.py#L81-L86

Changing this now (for EasyBuild v5.0) is worth considering.
There may be some fallout, but that can be handled via `exts_default_options`...

@mboisson If you would like to see this changes, please open a framework issue on this, so we can tag it with the `5.0` milestone.

Reply:
I don't *really* care whether extensions pick up parameters from the parent recipe, but I do know we need to be able to fix shebangs of extensions. This change is in fact already merged in our fork.

Reply:
Yeah, I agree.

And even if `fix_*_shebang` is being picked up from the parent once this change is merged, it should still be OK. Worst case, the shebangs get fixed...